src/httprpc.cpp

```diff
@@ -256,9 +250,24 @@ static bool InitRPCAuthentication()
         LogPrintf("Config options rpcuser and rpcpassword will soon be deprecated. Locally-run instances may remove rpcuser to use cookie-based auth, or may be replaced with rpcauth. Please see share/rpcauth for rpcauth auth generation.\n");
         strRPCUserColonPass = gArgs.GetArg("-rpcuser", "") + ":" + gArgs.GetArg("-rpcpassword", "");
     }
-    if (gArgs.GetArg("-rpcauth","") != "")
+    if (gArgs.GetArg("-rpcauth", "") != "" || gArgs.GetArg("-rpcauthfile", "") != "")
     {
         LogPrintf("Using rpcauth authentication.\n");
+        for (std::string rpcauth : gArgs.GetArgs("-rpcauth")) {
+            std::vector<std::string> fields;
+            boost::split(fields, rpcauth, boost::is_any_of(":$"));
+            if (fields.size() == 3) g_rpcauth.push_back(fields);
```

Comment:
I guess it's a preexisting problem, but lack of any error feedback when the syntax is wrong seems like a usability nightmare

Reply:
Yeah it is, can be improved.

Reply:
Actually I think I'll do it in this branch, `-rpcauth` values are now processed once.

Reply:
Such an `-rpccookiegenerate=false` option is what I had in mind in #20387. I would prefer that exactly for the simplicity of using an arbitrary string as cookie (which I can share via Docker's secret mechanism).

Reply:
Syntax isn't necessarily wrong. It could be a 4th field added (eg, Knots uses it to restrict access to the named wallet).

A debug log line can't hurt though.